packages/mendel-pipeline/src/pipeline.js

```diff
@@ -19,15 +19,16 @@ function MendelPipeline(options) {
 
     const registry = new MendelRegistry(options);
     const transformer = new Transformer(options.transforms, options);
+    const depsResolver = new DepResolver(options);
 
-    // Pipeline steps
-    const initializer = new Initialize({registry, transformer}, options);
-    const watcher = new Watcher({registry, transformer}, options);
-    const reader = new Reader({registry, transformer}, options);
-    const ist = new IST({registry, transformer}, options);
-    const depsResolver = new DepResolver({registry, transformer}, options);
+    const toolset = {registry, transformer, depsResolver};
```

Comment:
If we impose a restriction that every plugin has to operate on AST (potentially one type?) and output AST, then yes. If not, no.